views/js/runner/proxy/cache/proxy.js

```diff
@@ -63,27 +63,37 @@ define([
             var self = this;
 
             //we keep items here
-            this.itemStore    = itemStoreFactory(cacheSize);
-
-            //keep track of the calls to prevent doing it again on navigating back and forward
-            this.nextCalledBy = {};
+            this.itemStore = itemStoreFactory(cacheSize);
 
             //can we load the next item from the cache/store ?
-            this.getItemFromStore  = false;
-
-            //check whether we're on the last item
-            this.isLastItem   = false;
+            this.getItemFromStore = false;
 
             //keep a ref to the promise of the loadNextItem in case the call is not done when moving
             this.loadNextPromise = Promise.resolve();
 
+            //preload at least this number of items
+            this.cacheAmount = 1;
+
+            //maintain a map of the items chain: for each item, a link to its neighbors
+            this.itemChain = {};
```

Comment:
You are right! I could also make that from the map builder, server side, aside the list of jumps. That way there should be only one place where we compute the map.